### ISSUE TITLE
REDIS BUG REPORT - Accessing address: (nil)

### ISSUE THREAD

**Issue by avifatal** [RedisGraph/RedisGraph#1281]
Hi
I have the following error happening in production with docker 2.2.3 and edge. the odd thing is, it is working fin on a local docker.
I assume that there might be something with the data.
Can this stack give me a hint?

Thanks
 
```
=== REDIS BUG REPORT START: Cut & paste starting from here ===
1:M 13 Aug 2020 14:37:10.442 # Redis 6.0.5 crashed by signal: 11
1:M 13 Aug 2020 14:37:10.442 # Crashed running the instruction at: 0x7f823b4fd6ca
1:M 13 Aug 2020 14:37:10.442 # Accessing address: (nil)
1:M 13 Aug 2020 14:37:10.442 # Failed assertion: <no assertion failed> (<no file>:0)

------ STACK TRACE ------
EIP:
/lib/x86_64-linux-gnu/libc.so.6(+0x1576ca)[0x7f823b4fd6ca]

Backtrace:
redis-server *:6379(logStackTrace+0x32)[0x561426f71e02]
redis-server *:6379(sigsegvHandler+0x9e)[0x561426f724de]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x12730)[0x7f823b57b730]
/lib/x86_64-linux-gnu/libc.so.6(+0x1576ca)[0x7f823b4fd6ca]
/usr/lib/redis/modules/redisgraph.so(_GraphContext_GetLabelID+0x55)[0x7f823a3190c5]
/usr/lib/redis/modules/redisgraph.so(GraphContext_GetSchema+0x10)[0x7f823a319170]
/usr/lib/redis/modules/redisgraph.so(GraphContext_GetIndex+0x12)[0x7f823a319542]
/usr/lib/redis/modules/redisgraph.so(reduce_scan_op+0x40)[0x7f823a317660]
/usr/lib/redis/modules/redisgraph.so(utilizeIndices+0x6f)[0x7f823a317bef]
/usr/lib/redis/modules/redisgraph.so(_optimizePlan+0x11)[0x7f823a3169c1]
/usr/lib/redis/modules/redisgraph.so(ExecutionPlan_PreparePlan+0x18)[0x7f823a305e08]
/usr/lib/redis/modules/redisgraph.so(Graph_Query+0x34c)[0x7f823a30077c]
/usr/lib/redis/modules/redisgraph.so(+0x1ea51d)[0x7f823a33151d]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x7fa3)[0x7f823b570fa3]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x3f)[0x7f823b49f4cf]

------ INFO OUTPUT ------
# Server
redis_version:6.0.5
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:db63ea56716d515f
redis_mode:standalone
os:Linux 5.3.0-1030-gcp x86_64
arch_bits:64
multiplexing_api:epoll
atomicvar_api:atomic-builtin
gcc_version:8.3.0
process_id:1
run_id:f754540c0dd326c540446f6d287504f540bd73d6
tcp_port:6379
uptime_in_seconds:30
uptime_in_days:0
hz:10
configured_hz:10
lru_clock:3493910
executable:/data/redis-server
config_file:

# Clients
connected_clients:1
client_recent_max_input_buffer:2
client_recent_max_output_buffer:0
blocked_clients:1
tracking_clients:0
clients_in_timeout_table:0

# Memory
used_memory:4652416
used_memory_human:4.44M
used_memory_rss:17289216
used_memory_rss_human:16.49M
used_memory_peak:4652416
used_memory_peak_human:4.44M
used_memory_peak_perc:101.23%
used_memory_overhead:854930
used_memory_startup:837872
used_memory_dataset:3797486
used_memory_dataset_perc:99.55%
allocator_allocated:4741928
allocator_active:5173248
allocator_resident:7729152
total_system_memory:16790847488
total_system_memory_human:15.64G
used_memory_lua:37888
used_memory_lua_human:37.00K
used_memory_scripts:0
used_memory_scripts_human:0B
number_of_cached_scripts:0
maxmemory:0
maxmemory_human:0B
maxmemory_policy:noeviction
allocator_frag_ratio:1.09
allocator_frag_bytes:431320
allocator_rss_ratio:1.49
allocator_rss_bytes:2555904
rss_overhead_ratio:2.24
rss_overhead_bytes:9560064
mem_fragmentation_ratio:3.80
mem_fragmentation_bytes:12736048
mem_not_counted_for_evict:0
mem_replication_backlog:0
mem_clients_slaves:0
mem_clients_normal:16986
mem_aof_buffer:0
mem_allocator:jemalloc-5.1.0
active_defrag_running:0
lazyfree_pending_objects:0

# Persistence
loading:0
rdb_changes_since_last_save:10470
rdb_bgsave_in_progress:0
rdb_last_save_time:1597329400
rdb_last_bgsave_status:ok
rdb_last_bgsave_time_sec:-1
rdb_current_bgsave_time_sec:-1
rdb_last_cow_size:0
aof_enabled:1
aof_rewrite_in_progress:0
aof_rewrite_scheduled:0
aof_last_rewrite_time_sec:-1
aof_current_rewrite_time_sec:-1
aof_last_bgrewrite_status:ok
aof_last_write_status:ok
aof_last_cow_size:0
module_fork_in_progress:0
module_fork_last_cow_size:0
aof_current_size:9315655
aof_base_size:9315655
aof_pending_rewrite:0
aof_buffer_length:0
aof_rewrite_buffer_length:0
aof_pending_bio_fsync:0
aof_delayed_fsync:0

# Stats
total_connections_received:1
total_commands_processed:3
instantaneous_ops_per_sec:0
total_net_input_bytes:507
total_net_output_bytes:7607
instantaneous_input_kbps:0.00
instantaneous_output_kbps:0.00
rejected_connections:0
sync_full:0
sync_partial_ok:0
sync_partial_err:0
expired_keys:0
expired_stale_perc:0.00
expired_time_cap_reached_count:0
expire_cycle_cpu_milliseconds:0
evicted_keys:0
keyspace_hits:20940
keyspace_misses:1
pubsub_channels:0
pubsub_patterns:0
latest_fork_usec:0
migrate_cached_sockets:0
slave_expires_tracked_keys:0
active_defrag_hits:0
active_defrag_misses:0
active_defrag_key_hits:0
active_defrag_key_misses:0
tracking_total_keys:0
tracking_total_items:0
tracking_total_prefixes:0
unexpected_error_replies:0

# Replication
role:master
connected_slaves:0
master_replid:02bb943235cad6887a7ee47a25e162f6657b366d
master_replid2:0000000000000000000000000000000000000000
master_repl_offset:0
second_repl_offset:-1
repl_backlog_active:0
repl_backlog_size:1048576
repl_backlog_first_byte_offset:0
repl_backlog_histlen:0

# CPU
used_cpu_sys:0.531390
used_cpu_user:16.132945
used_cpu_sys_children:0.002356
used_cpu_user_children:0.003204

# Modules
module:name=graph,ver=20019,api=1,filters=0,usedby=[],using=[],options=[]

# Commandstats
cmdstat_info:calls=2,usec=198,usec_per_call=99.00
cmdstat_graph.QUERY:calls=1,usec=43,usec_per_call=43.00

# Cluster
cluster_enabled:0

# Keyspace
db0:keys=1,expires=0,avg_ttl=0

------ CLIENT LIST OUTPUT ------
id=5 addr=172.27.0.4:54698 fd=10 name= age=30 idle=0 flags=b db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=32768 obl=0 oll=0 omem=0 events=r cmd=graph.QUERY user=default

------ REGISTERS ------
1:M 13 Aug 2020 14:37:10.445 #
RAX:00000000000009e0 RBX:0000000000000000
RCX:0000000000000001 RDX:0000000000000000
RDI:0000000000000000 RSI:00007f823af229e0
RBP:0000000000000000 RSP:00007f8239942d98
R8 :0000000000000033 R9 :00007f8234e070d8
R10:00007f8239943ee0 R11:0000000000000006
R12:0000000000000003 R13:00007f823ae3528c
R14:00007f8234e2c240 R15:00007f8234e11070
RIP:00007f823b4fd6ca EFL:0000000000010287
CSGSFS:002b000000000033
1:M 13 Aug 2020 14:37:10.445 # (00007f8239942da7) -> 0000000000000000
1:M 13 Aug 2020 14:37:10.445 # (00007f8239942da6) -> 00007f823a317660
1:M 13 Aug 2020 14:37:10.445 # (00007f8239942da5) -> 00007f8234e1f050
1:M 13 Aug 2020 14:37:10.445 # (00007f8239942da4) -> 0000000000000000
1:M 13 Aug 2020 14:37:10.445 # (00007f8239942da3) -> 00007f8234e460a0
1:M 13 Aug 2020 14:37:10.445 # (00007f8239942da2) -> 00007f823a319542
1:M 13 Aug 2020 14:37:10.445 # (00007f8239942da1) -> 0000000000000001
1:M 13 Aug 2020 14:37:10.445 # (00007f8239942da0) -> 0000000000000000
1:M 13 Aug 2020 14:37:10.445 # (00007f8239942d9f) -> 00007f8234e1f050
1:M 13 Aug 2020 14:37:10.445 # (00007f8239942d9e) -> 00007f823a319170
1:M 13 Aug 2020 14:37:10.445 # (00007f8239942d9d) -> 00007f8234e1900c
1:M 13 Aug 2020 14:37:10.445 # (00007f8239942d9c) -> 00007f8234e1901c
1:M 13 Aug 2020 14:37:10.445 # (00007f8239942d9b) -> 0000000000000000
1:M 13 Aug 2020 14:37:10.445 # (00007f8239942d9a) -> 00007f823ae22000
1:M 13 Aug 2020 14:37:10.445 # (00007f8239942d99) -> 00007f823a9ff1e8
1:M 13 Aug 2020 14:37:10.445 # (00007f8239942d98) -> 00007f823a3190c5

------ MODULES INFO OUTPUT ------

------ FAST MEMORY TEST ------
1:M 13 Aug 2020 14:37:10.446 # Bio thread for job type #0 terminated
1:M 13 Aug 2020 14:37:10.446 # Bio thread for job type #1 terminated
1:M 13 Aug 2020 14:37:10.446 # Bio thread for job type #2 terminated
*** Preparing to test memory region 5614270bd000 (2277376 bytes)
*** Preparing to test memory region 561427a56000 (3506176 bytes)
*** Preparing to test memory region 7f8228000000 (135168 bytes)
*** Preparing to test memory region 7f8230000000 (2146304 bytes)
*** Preparing to test memory region 7f8234600000 (8388608 bytes)
*** Preparing to test memory region 7f8234e00000 (4194304 bytes)
*** Preparing to test memory region 7f82353be000 (5767168 bytes)
*** Preparing to test memory region 7f823593f000 (8388608 bytes)
*** Preparing to test memory region 7f8236140000 (8388608 bytes)
*** Preparing to test memory region 7f8236941000 (8388608 bytes)
*** Preparing to test memory region 7f8237142000 (8388608 bytes)
*** Preparing to test memory region 7f8237943000 (8388608 bytes)
*** Preparing to test memory region 7f8238144000 (8388608 bytes)
*** Preparing to test memory region 7f8238945000 (8388608 bytes)
*** Preparing to test memory region 7f8239146000 (8388608 bytes)
*** Preparing to test memory region 7f8239947000 (8388608 bytes)
*** Preparing to test memory region 7f823a9fd000 (12288 bytes)
*** Preparing to test memory region 7f823aa00000 (8388608 bytes)
*** Preparing to test memory region 7f823b3a2000 (16384 bytes)
*** Preparing to test memory region 7f823b563000 (24576 bytes)
*** Preparing to test memory region 7f823b586000 (16384 bytes)
*** Preparing to test memory region 7f823b86f000 (16384 bytes)
*** Preparing to test memory region 7f823ba97000 (8192 bytes)
*** Preparing to test memory region 7f823bac4000 (4096 bytes)
.O.O.O.O.O.O.O.O.O.O.O.O.O.O.1:C 13 Aug 2020 14:37:12.089 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 13 Aug 2020 14:37:12.089 # Redis version=6.0.5, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 13 Aug 2020 14:37:12.089 # Configuration loaded
1:M 13 Aug 2020 14:37:12.090 * Running mode=standalone, port=6379.
1:M 13 Aug 2020 14:37:12.091 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
1:M 13 Aug 2020 14:37:12.091 # Server initialized
1:M 13 Aug 2020 14:37:12.091 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
1:M 13 Aug 2020 14:37:12.091 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
1:M 13 Aug 2020 14:37:12.099 * <graph> Graph deletion will be done asynchronously.
1:M 13 Aug 2020 14:37:12.099 * <graph> Thread pool created, using 4 threads.
1:M 13 Aug 2020 14:37:12.099 * <graph> Maximum number of OpenMP threads set to 4
1:M 13 Aug 2020 14:37:12.100 * Module 'graph' loaded from /usr/lib/redis/modules/redisgraph.so
1:M 13 Aug 2020 14:37:26.895 * DB loaded from append only file: 14.795 seconds
1:M 13 Aug 2020 14:37:26.895 * Ready to accept connections
1:signal-handler (1597329447) Received SIGTERM scheduling shutdown...
1:M 13 Aug 2020 14:37:27.296 # User requested shutdown...
1:M 13 Aug 2020 14:37:27.296 * Calling fsync() on the AOF file.
1:M 13 Aug 2020 14:37:27.297 # Redis is now ready to exit, bye bye...
```

**Comment by swilly22** [comment]
@avifatal, do you know which query triggered this crash?

**Comment by avifatal** [comment]
I suspect that this query makes it.
Again, this only in production.
Thanks
```
MATCH (n:GraphTarget {AppId:$AppId})-[oo:OwnerOf]->(a0:Source)-[fo:FriendOf]-(a1)-[fo1:FriendOf*1..3]-(a2)<-[oo2:OwnerOf]-(z:GraphTarget)
      WHERE a0.WebProfile <> 'INTERNET' AND a1.WebProfile <> 'INTERNET' AND a2.WebProfile <> 'INTERNET' AND a1 <> a0 AND a2 <> a1 AND a2 <> a0 and n <> z
      AND (a1:Source)-[:FriendOf*1..3]-(a2:Source)<-[oo2]-(z)
      return distinct(a1),z,fo.Verified
```

**Comment by jeffreylovitz** [comment]
Hi @avifatal,

I was able to reproduce the issue using your query, and should have a fix out soon. The crash is caused (strangely) by specifying the node labels in your final path filter, so as a workaround, you can rewrite your query like so:
```
MATCH (n:GraphTarget {AppId:$AppId})-[oo:OwnerOf]->(a0:Source)-[fo:FriendOf]-(a1:Source)-[fo1:FriendOf*1..3]-(a2:Source)<-[oo2:OwnerOf]-(z:GraphTarget)
      WHERE a0.WebProfile <> 'INTERNET' AND a1.WebProfile <> 'INTERNET' AND a2.WebProfile <> 'INTERNET' AND a1 <> a0 AND a2 <> a1 AND a2 <> a0 and n <> z
      AND (a1)-[:FriendOf*1..3]-(a2)<-[oo2]-(z)
      return distinct(a1),z,fo.Verified
```

**Comment by avifatal** [comment]
Hi @jeffreylovitz 
Please see the images below. On the local, your query brings different results than my query., (again, production has an error).
Your query:

![image](https://user-images.githubusercontent.com/12430740/90213006-6c337f00-ddfd-11ea-9d61-1e817afcfde6.png)

This is my query.
Brings different results....

![image](https://user-images.githubusercontent.com/12430740/90213020-735a8d00-ddfd-11ea-8ec3-da2bd4bb9ba2.png)

**Comment by jeffreylovitz** [comment]
Hi @avifatal,

#1283 should resolve this crash. The different results are due to a bug that is not yet fixed, which is being tracked in #1284.

I believe that my query above is the one which produces correct results.
